src/components/NewClusterModal.js

```diff
@@ -211,27 +211,35 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     return { size: newPersistentDisk.size }
   }
 
-  sendDeleteMetrics(isDeleteRuntime, isDeletePersistentDisk) {
-    Ajax().Metrics.captureEvent(Events.cloudEnvironmentDelete, {
-      ...extractWorkspaceDetails(this.makeWorkspaceObj()),
-      isDeleteRuntime,
-      isDeletePersistentDisk
-    })
-  }
-
-  sendUpdateMetrics(newRuntime, newPersistentDisk) {
-    const machineType = newRuntime.cloudService === cloudServices.GCE ? newRuntime.machineType : newRuntime.masterMachineType
-    const { cpu, memory } = findMachineType(machineType)
+  sendCloudEnvironmentMetrics() {
+    const metricsEvent = Utils.cond(
+      [(this.state.viewMode === 'deleteEnvironmentOptions'), () => 'cloudEnvironmentDelete'],
+      [(!!this.getOldEnvironmentConfig().runtime), () => 'cloudEnvironmentUpdate'],
+      () => 'cloudEnvironmentCreate'
+    )
+    const { runtime: newRuntime, persistentDisk: newPersistentDisk } = this.getNewEnvironmentConfig()
+    const { runtime: oldRuntime, persistentDisk: oldPersistentDisk } = this.getOldEnvironmentConfig()
+    const newMachineType = newRuntime && (newRuntime.cloudService === cloudServices.GCE ? newRuntime.machineType : newRuntime.masterMachineType)
+    const oldMachineType = oldRuntime && (oldRuntime?.cloudService === cloudServices.GCE ? oldRuntime.machineType : oldRuntime.masterMachineType)
+    const { cpu: newRuntimeCpus, memory: newRuntimeMemory } = findMachineType(newMachineType)
+    const { cpu: oldRuntimeCpus, memory: oldRuntimeMemory } = findMachineType(oldMachineType)
 
-    Ajax().Metrics.captureEvent(Events.cloudEnvironmentUpdate, {
+    Ajax().Metrics.captureEvent(Events[metricsEvent], {
       ...extractWorkspaceDetails(this.makeWorkspaceObj()),
-      ...newRuntime,
-      cpu,
-      memory,
-      persistentDiskSize: newPersistentDisk?.size,
-      runtimeCostPerHour: Utils.formatUSD(runtimeConfigCost(this.getPendingRuntimeConfig())),
-      runtimePausedCostPerHour: Utils.formatUSD(ongoingCost(this.getPendingRuntimeConfig())),
-      persistentDiskCostPerMonth: (newPersistentDisk && Utils.formatUSD(persistentDiskCostMonthly(newPersistentDisk)))
+      ..._.mapKeys(key => `newRuntime_${key}`, newRuntime),
+      newRuntime_exists: !!newRuntime,
+      newRuntime_cpus: newRuntime && newRuntimeCpus,
```

Comment:
Will these populate in mixpanel as undefined in delete events?

Reply:
No, if it's undefined it just won't populate in mixpanel at all, so the whole field won't show up in the mixpanel event. I think that's an acceptable action, rather than forcing everything to be false or something if it's not populated.